### DISCUSSION TITLE
Custom extrude lengths in extrude panel

### DISCUSSION THREAD

**Discussion by MakManIfest** [KlipperScreen/KlipperScreen#387] [Q&A]
I want to change the default extrude values in the extrude panel.

I tried editing this section in the extrude.py file,

"class ExtrudePanel(ScreenPanel):
    distance = 5
    distances = ['5', '10', '15', '25']"

to the values I want,

"class ExtrudePanel(ScreenPanel):
    distance = 5
    distances = ['1', '3', '10', '20']"

but got an error when loading the panel.

Something to do with the "distance = 5" line I assume.

Some help would be greatly appreciated.

Thanks!

**Comment by jordanruthe** [comment]
The `distance` variable is the default distance. You must change it to be an option that is in distances, so for your case:
```
distance = 1
distances = ['1', '3', '10', '20']"
```

There are some other places where you have to modify it. I'll make an update to make this easier in the near future.

**Reply by MakManIfest** [reply]
So do I like need to report an issue with this problem? It's been 8 months and no resolution. I mean it's an issue with basic functionality. Kinda needs to be sorted.

**Comment by MakManIfest** [comment]
Thanks!
After some fiddling I found the issue is with the 1st preset value. I think it is because its the preset selected when Klipperscreen start. Its the same with the extrude speed presets. The 2nd preset causes an error when I change it and it is also the one selected when Klipperscreen starts.
Otherwise, changing the distance variable causes the distance preset buttons to be un-selectable for some reason. When I tap on the 2nd, 3rd or 4th distance preset button it flashes but the first preset remains selected.

**Comment by MakManIfest** [comment]
[KlipperScreen.log](https://github.com/jordanruthe/KlipperScreen/files/7791337/KlipperScreen.log)

here is the log wit the error I get.

**Comment by MakManIfest** [comment]
console is reporting some funky numbers as well. Not sure if this is a separate issue or because I've changed the extrude presets.
Console in Mainsail is reporting the correct commands issued via klipperscreen but the console on klipperscreen is not.

**Comment by eboston** [comment]
Any updates on this?

**Comment by alfrix** [comment]
this is already possible https://klipperscreen.readthedocs.io/en/latest/Configuration/#printer-options
